core/remotes/resolver.go

```diff
@@ -43,7 +47,7 @@ type Resolver interface {
 	// Fetcher returns a new fetcher for the provided reference.
 	// All content fetched from the returned fetcher will be
 	// from the namespace referred to by ref.
-	Fetcher(ctx context.Context, ref string) (Fetcher, error)
+	Fetcher(ctx context.Context, ref string, opts ...FetcherOpt) (Fetcher, error)
```

Comment:
I don't think we should make this interface change. I was trying to think of some better options. Previously we had another interface appended with `WithOptions`, but I don't think that should be necessary here. This configuration  could just be directly given as resolver options. The resolver could have a `SetOptions` function defined and type-asserted to in the transfer case. I think we can avoid changing the interface here though, especially since these settings are global and not defined per call.

Reply:
Ack ! Updating !

Reply:
Okay, updated, thanks for the review, I think this looks better, WDYT ?